java/flight/src/main/java/org/apache/arrow/flight/FlightDescriptor.java

```diff
@@ -92,6 +94,28 @@ public boolean isCommand() {
     return b.setType(DescriptorType.PATH).addAllPath(path).build();
   }
 
+  /**
+   * Get the serialized form of this protocol message.
+   *
+   * <p>Intended to help interoperability by allowing non-Flight services to still return Flight types.
+   */
+  public byte[] toByteArray() {
+    return toProtocol().toByteArray();
+  }
+
+  /**
+   * Parse the serialized form of this protocol message.
+   *
+   * <p>Intended to help interoperability by allowing Flight clients to obtain stream info from non-Flight services.
+   *
+   * @param serialized The serialized form of the FlightDescriptor, as returned by {@link #toByteArray()}.
+   * @return The deserialized FlightDescriptor.
+   * @throws IOException if the serialized form is invalid.
+   */
+  public static FlightDescriptor fromByteArray(byte[] serialized) throws IOException {
```

Comment:
Random thought... what do you think about using bytebuffer in these methods (or bytebuf)? It makes things more consistent with the rest of Arrow.

Reply:
I'm happy to use ByteBuffer for consistency, it shouldn't make much of a difference either way.